### PR TITLE
WIP: Add executor for managing B.A.T.M.A.N. adv. interfaces

### LINKED ISSUE #45: Add support for B.A.T.M.A.N. adv. interfaces
Add support for manage B.A.T.M.A.N. adv. with ifupdown-ng

### PR DESCRIPTION
**Pull request by BarbarossaTM**
Closes #45

### CODE REVIEW

File: executor-scripts/linux/batman [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+#
+# Maximilian Wilhelm <max@sdn.clinic>
+#  --  Wed 26 Aug 2020 08:15:58 PM CEST
+#
+# This executor is responsible for setting up the main B.A.T.M.A.N. adv. interface (eg. bat0)
+# as well as managing settings of the underlying interfaces (hardifs).
+#
+# Know options for the main interface are:
```
Comment (kaniini) [inline executor-scripts/linux/batman:9]: Typo.

File: executor-scripts/linux/batman [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+#
+# Functions to manage main B.A.T.M.A.N. adv. interface
+batctl_if () {
+	for iface in ${IF_BATMAN_IFACES}; do
+		${MOCK} batctl "${mesh_if_param}" "${IFACE}" interface "$1" "${iface}"
+	done
+}
+
+set_batmam_params () {
```
Comment (kaniini) [inline executor-scripts/linux/batman:50]: shouldn't this be `set_batman_params()` ?

File: executor-scripts/linux/batman [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	depend)
+		if [ "${IF_BATMAN_IFACES}" ]; then
+			echo "${IF_BATMAN_IFACES}"
+		fi
+		;;
+
+	pre-up)
+		# Main B.A.T.M.A.N. adv. interface
+		if [ "${IF_BATMAN_IFACES}" ]; then
+			batctl "${mesh_if_param}" "${IFACE}" interface create || true
```
Comment (kaniini) [inline executor-scripts/linux/batman:74]: why `|| true` at the end?  is there some reason we would want to suppress failure?

**Review by kaniini — CHANGES_REQUESTED** [review]

File: executor-scripts/linux/batman (hunk 3 above)
Comment (BarbarossaTM) [reply executor-scripts/linux/batman]: That's compatiblity glue for older version of `batctl` which don't have the `create` call and you have to add member interfaces to create the batman interface

File: executor-scripts/linux/batman (hunk 2 above)
Comment (BarbarossaTM) [reply executor-scripts/linux/batman:50]: Hm, it works with /bin/sh being dash on Debian I think. Am I violating POSIX? :)

**Comment by kaniini** [comment]
looks ok to me.

**Review by kaniini — APPROVED** [review]

**Pull request merged**